tools/Voron.Recovery/Recovery.cs

```diff
@@ -496,6 +533,31 @@ void PrintRecoveryProgress(long startOffset, byte* mem, byte* eof, DateTime now)
             }
         }
 
+        private void CheckInvalidPagesInARaw(PageHeader* pageHeader, byte* mem)
```

Comment:
Just to make sure.. what did you mean by "in a raw" ? I see you count it so maybe you meant "in a row" but from the other side I don't see you reset `_InvalidChecksumWithNoneZeroMac` anywhere

Reply:
It is only the first pages in a row, once a valid page checksum was found we disable this check by using the flag: _shouldIgnoreInvalidPagesInARaw